### PR TITLE
YouTube video capture and annotation

### PR DESCRIPTION
**Pull request by LukasGibb**
Fixes: https://github.com/LukasGibb/falcon-tracker/issues/8

comparer-function:
- removed unused imports
- publish video URL via pubsub
- better memory management
- better error handling

annotation-function:
- add code to download clip from youtube video
- add code to send clip off to the API for annotation
- made script work with pub/sub trigger or http trigger for flexibility and local testing
- add variable to set length (sec) of CAPTURE_TIME
- set function timeout to allow for CAPTURE_TIME

### CODE REVIEW

**Review by andrewirvingeagleeye — APPROVED** [review]

**Comment by arivictor** [comment]
deploying

**Comment by arivictor** [comment]
We might need to bump up the `timeout` on the annotation function as it seems to get close or exceed it at times.

File: cloud-functions/annotation-function/main.py [hunk 1]
```diff
@@ -1,19 +1,54 @@
...
+import io
+import base64
 
 import functions_framework
 
 @functions_framework.http
 def handle(request):
     # This function should call the Google Vision/Video API and return
-    
+    """handle a request to the function"""
```
Comment (arivictor) [inline cloud-functions/annotation-function/main.py:15]: Update the terraform for this function to increase `timeout`

**Review by arivictor — CHANGES_REQUESTED** [review]

File: cloud-functions/annotation-function/main.py (hunk 1 above)
Comment (LukasGibb) [reply cloud-functions/annotation-function/main.py:15]: Added another commit to address this. Timeout is now set dynamically to allow time to capture and analyse the video.

**Review by arivictor — APPROVED** [review]

**Pull request merged**